cg/store/crud/read.py

```diff
@@ -1430,3 +1430,37 @@ def get_order_by_ticket_id(self, ticket_id: int) -> Order | None:
             orders=orders, filters=order_filter_functions, ticket_id=ticket_id
         )
         return orders.first()
+
+    def get_not_received_case_count(self, order_id: int) -> int:
+        filters: list[CaseFilter] = [CaseFilter.BY_ORDER, CaseFilter.NOT_RECEIVED]
+        cases: Query = self._get_join_case_and_sample_query()
+        return apply_case_filter(
+            cases=cases,
+            filter_functions=filters,
+            order_id=order_id,
+        ).count()
```

Comment:
I think this makes a lot of sense to move the "heavy" lifting to the db as it is designed to do this type of stuff efficiently.

Reply:
I'll measure just to make sure, but it should be much faster this way!